docs/tutorials/particle-filtering.md

```diff
@@ -1,3 +1,564 @@
 # Primer on Particle Filtering with Beluga
 
-Coming soon!
+:::{figure} ../_images/beluga_tutorial.gif
+:alt: Short video of a beluga_tutorial's record.
+:::
+
+## Overview
+
+Modern robotic systems operate in unstructured, unpredictable environments, making robust software crucial to handle diverse situations.
+A key challenge is uncertainty, which occurs when robots lack critical task information. Probabilistic robotics addresses this by
+representing uncertainty explicitly through probability theory, rather than relying on a single best guess.
+
+The robot localization problem is one of the most important ones in mobile robotics, as it is a fundamental requirement for autonomous navigation. The Monte Carlo Localization (MCL) algorithm is one of the most widely used algorithms for map-based localization and has become a key component in many robotic systems [^BelugaPaper]. **TODO: It is necessary to put here the reference, or just include it at the end?**
+
+
+The purpose of this tutorial is to show how to implement a custom {abbr}`MCL (Monte Carlo Localiaztion)` algorithm using **Beluga**,
+motivated by the example given by _Probabilistic Robotics [^ProbRob], page 200, figure 8.11_. **TODO: Can I used the image of the book?**
+
+The code is written using C++17 with the ranges-v3 [^RangesV3] library for range handling along with its respective algorithms.
+
+## Requirements
+
+Before starting the tutorial, first [install Beluga](../getting-started/installation.md).
+
+:::{important}
+I highly recommend reading the [key concepts](../concepts/key-concepts.md) section before diving into this tutorial.
+:::
+
+## Tasks
```

Comment:
I think this is missing the problem formulation at the beginning. Can we describe what this 1D world looks like, how do we plan to model the motion of the robot and what are the characteristics of the sensor before diving into the code?

I think some of the explanation done below should be moved here.